server/apps/research/serializers/article_serializer.py

```
@@ -59,24 +95,29 @@
                 article.authors.set(authors)
             if categories:
                 article.categories.set(categories)
+            if related_articles is not None:
+                article.related_articles.set(related_articles)
 
             return article
         except Exception as e:            
             raise serializers.ValidationError(f"Error creating article: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)